consensus/hotstuff/signature/consensus_test.go

```diff
@@ -0,0 +1,49 @@
+package signature
+
+import (
+	"testing"
+
+	"github.com/onflow/flow-go/consensus/hotstuff"
+	"github.com/onflow/flow-go/utils/unittest"
+	"github.com/stretchr/testify/require"
+)
+
+func TestEncodeDecodeStakingSig(t *testing.T) {
+	sig := unittest.SignatureFixture()
+	encoded := EncodeSingleSig(hotstuff.SigTypeStaking, sig)
+	decodedType, decodedSig, err := DecodeSingleSig(encoded)
+	require.NoError(t, err)
+	require.Equal(t, hotstuff.SigTypeStaking, decodedType)
+	require.Equal(t, sig, decodedSig)
+}
+
+func TestEncodeDecodeRandomBeaconSig(t *testing.T) {
+	sig := unittest.SignatureFixture()
+	encoded := EncodeSingleSig(hotstuff.SigTypeRandomBeacon, sig)
+	decodedType, decodedSig, err := DecodeSingleSig(encoded)
+	require.NoError(t, err)
+	require.Equal(t, hotstuff.SigTypeRandomBeacon, decodedType)
+	require.Equal(t, sig, decodedSig)
+}
```

Comment:
We could merge these 2 tests into one test `TestEncodeDecodeSig` using something like :
```
types := []hotstuff.SigType { hotstuff.SigTypeRandomBeacon, hotstuff.SigTypeStaking} 
for _, t := range types {
...
}
```